src/PrestaShopBundle/Translation/Provider/UserTranslatedCatalogueProvider.php

```
@@ -0,0 +1,91 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShopBundle\Translation\Loader\DatabaseTranslationLoader;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Gets catalogue translated by the user himself ans stored in the database.
+ */
+class UserTranslatedCatalogueProvider implements TranslationCatalogueProviderInterface
+{
+    /**
+     * @var DatabaseTranslationLoader
+     */
+    private $databaseTranslationReader;
+
+    /**
+     * @var array
+     */
+    private $translationDomains;
+    /**
+     * @var string|null
+     */
+    private $themeName;
+
+    /**
+     * @param DatabaseTranslationLoader $databaseTranslationReader
+     * @param array $translationDomains
+     * @param string|null $themeName
+     */
+    public function __construct(
+        DatabaseTranslationLoader $databaseTranslationReader,
+        array $translationDomains,
+        ?string $themeName = null
+    ) {
+        $this->databaseTranslationReader = $databaseTranslationReader;
+        $this->translationDomains = $translationDomains;
+        $this->themeName = $themeName;
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return MessageCatalogue
+     */
+    public function getCatalogue(string $locale): MessageCatalogue
+    {
+        $catalogue = new MessageCatalogue($locale);
+
+        foreach ($this->translationDomains as $translationDomain) {
+            $domainCatalogue = $this->databaseTranslationReader->load(
+                null,
+                $locale,
+                $translationDomain,
+                $this->themeName
+            );
+
+            if ($domainCatalogue instanceof MessageCatalogue) {
+                $catalogue->addCatalogue($domainCatalogue);
+            }
+        }
+
+        return $catalogue;
```

If `$this->translationDomains` is empty or if `$domainCatalogue` is not an instance of MessageCatalogue, then what is returned here is `new MessageCatalogue($locale);`

Is that OK? Is that expected?
If it's not OK, maybe we can throw an exception, like `CouldNotFindMessageCatalogueException`? I'm thinking about [crashing rather than providing default broken behavior](https://www.quora.com/What-does-crash-early-mean-in-software-engineering).

---

Returning an empty DB catalogue wont break anything. If the user never change any translation, the result will be empty too

---

But I will check that translationsDOmains are strings only ;)